File: CareConnect/CareConnect.Models/Requests/ChildUpdateRequest.cs
using System;
using System.Text.Json.Serialization;

namespace CareConnect.Models.Requests
{
    public class ChildUpdateRequest
    {
        public string? FirstName { get; set; } 

        public string? LastName { get; set; } 

        public DateTime? BirthDate { get; set; }

        [JsonIgnore]
        public DateTime ModifiedDate { get; set; } = DateTime.Now;
    }   
}

File: UI/careconnect_mobile/lib/models/requests/child_update_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'child_update_request.g.dart';

@JsonSerializable()
class ChildUpdateRequest {
  String? firstName;
  String? lastName;
  DateTime? birthDate;

  ChildUpdateRequest({
    required this.firstName,
    required this.lastName,
    required this.birthDate,
  });

  factory ChildUpdateRequest.fromJson(Map<String, dynamic> json) =>
      _$ChildUpdateRequestFromJson(json);

  Map<String, dynamic> toJson() => _$ChildUpdateRequestToJson(this);
}
